File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    var stack = [];
    for (let i = 0; i < s.length+1; i++) {
        while (stack.length > 0) {
            if (stack[stack.length - 1] === ")" && stack[stack.length - 2] === "(") {
                stack.pop();
                stack.pop();
            } else {
                break;
            }
        }
        stack.push(s[i]);
    }
    if (stack.length !== 1 && stack[0] !== undefined) {
        answer = false;
    }
    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stk = []
    for i in s:
        stk.append(i)
        if len(stk) > 1 and stk[-1] == ')' and stk[-2] == '(':
            stk.pop()
            stk.pop()
    if len(stk) == 0:
        return True
    else:
        return False